### PR TITLE
[SPARK-8032] [PySpark] Make version checking for NumPy in MLlib more robust

### PR DESCRIPTION
**Pull request by MechCoder**
The current checking does version `1.x' is less than`1.4' this will fail if x has greater than 1 digit, since x > 4, however `1.x` < `1.4`

It fails in my system since I have version `1.10` :P

### CODE REVIEW

**Comment by MechCoder** [comment]
ping @mengxr @jkbradley

**Comment by MechCoder** [comment]
jenkins test this

File: python/pyspark/mllib/__init__.py [hunk 1]
```diff
@@ -23,7 +23,14 @@
 # MLlib currently needs NumPy 1.4+, so complain if lower
 
 import numpy
-if numpy.version.version < '1.4':
+ver = numpy.version.version
```
Comment (mengxr) [inline python/pyspark/mllib/__init__.py]: Thanks for catching this before the official 1.10 release comes out! The following may be simpler:

~~~python
ver = [int(x) for x in numpy.version.version.split(.)[:2]]
if ver < [1, 4]:
  raise ...
~~~

Comment (MechCoder) [inline python/pyspark/mllib/__init__.py]: Is this way of list comparison common? Funny, I haven't used it before.

**Comment by MechCoder** [comment]
done

File: python/pyspark/mllib/__init__.py (hunk 1 above)
Comment (mengxr) [inline python/pyspark/mllib/__init__.py]: See the last section in https://docs.python.org/2/tutorial/datastructures.html

**Comment by mengxr** [comment]
@MechCoder Why it is `<=` instead of `<`? We do support numpy 1.4.x like 1.4.1.

**Comment by MechCoder** [comment]
I see, I misunderstood the comment '"MLlib requires NumPy 1.4+"

**Comment by mengxr** [comment]
@MechCoder please test it on your local and watch the Jenkins error message. For example, 

In https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/34003/consoleFull, you can see

~~~
Running test: pyspark.mllib.classification ... Traceback (most recent call last):
  File "/usr/lib64/python2.6/runpy.py", line 104, in _run_module_as_main
    loader, code, fname = _get_module_details(mod_name)
  File "/usr/lib64/python2.6/runpy.py", line 79, in _get_module_details
    loader = get_loader(mod_name)
  File "/usr/lib64/python2.6/pkgutil.py", line 456, in get_loader
    return find_loader(fullname)
  File "/usr/lib64/python2.6/pkgutil.py", line 466, in find_loader
    for importer in iter_importers(fullname):
  File "/usr/lib64/python2.6/pkgutil.py", line 422, in iter_importers
    __import__(pkg)
  File "pyspark/mllib/__init__.py", line 27
    ver = [int(x) for x in numpy.version.version.split(.)[:2]]
~~~

**Comment by MechCoder** [comment]
lol. sorry about the silly mistake.

**Comment by mengxr** [comment]
LGTM. Merged into master and all branches since 1.0. Thanks!

**Pull request closed without merging**